### PR TITLE
Made path, serializer and viewset for companies types

### PR DESCRIPTION
**Pull request by IegorSoboliev**

### CODE REVIEW

File: stats/views.py [hunk 1]
```diff
@@ -43,11 +45,16 @@ def get(self, request):
...
 
-class TopKvedsView(generics.ListAPIView):
+class TopKvedView(generics.ListAPIView):
     # permission_classes = [AllowAny]
+    not_valid_kved_id = Kved.objects.get(code='not_valid')
     queryset = (CompanyToKved.objects.select_related(
         'kved', 'kved__group', 'kved__section', 'kved__division',
-    ).values('kved').annotate(
+    ).exclude(
+        kved=not_valid_kved_id
```
Comment (roman-tiukh) [inline stats/views.py:54]:
```suggestion
        kved__code='not_valid'
```

File: stats/views.py [hunk 2]
```diff
@@ -43,11 +45,16 @@ def get(self, request):
         }, status=200)
 
 
-class TopKvedsView(generics.ListAPIView):
+class TopKvedView(generics.ListAPIView):
     # permission_classes = [AllowAny]
+    not_valid_kved_id = Kved.objects.get(code='not_valid')
```
Comment (roman-tiukh) [inline stats/views.py:50]: remove this line, getting objects in class body is bad idea

File: business_register/serializers/company_serializers.py [hunk 3]
```diff
@@ -78,7 +83,7 @@ class CompanySerializer(serializers.ModelSerializer):
     authorized_capital = serializers.FloatField()
     parent = serializers.StringRelatedField()
     predecessors = serializers.StringRelatedField(many=True)
-    company_type = serializers.StringRelatedField()
+    company_type = CompanyTypeSerializer()
```
Comment (roman-tiukh) [inline business_register/serializers/company_serializers.py:86]: why? i think its bad idea

File: stats/views.py [hunk 4]
```diff
@@ -56,3 +63,25 @@ class TopKvedsView(generics.ListAPIView):
...
+        'company_type'
+    ).annotate(
+        count_types=Count('company_type')
+    ).order_by('-count_types'))
+    serializer_class = TopCompanyTypeSerializer
+    pagination_class = None
+
+    @method_decorator(cache_page(60 * 60 * 24))
+    def get(self, request, *args, **kwargs):
+        return super().get(request, *args, **kwargs)
```
Comment (roman-tiukh) [inline stats/views.py:87]: need to rework

File: stats/views.py (hunk 1 above)
Comment (IegorSoboliev) [reply stats/views.py:54]: thank you, done!

File: stats/views.py (hunk 2 above)
Comment (IegorSoboliev) [reply stats/views.py:50]: definitely)

File: stats/views.py [hunk 5]
```diff
@@ -43,16 +44,35 @@ def get(self, request):
         }, status=200)
 
 
-class TopKvedsView(generics.ListAPIView):
-    # permission_classes = [AllowAny]
+class TopKvedView(generics.ListAPIView):
+    permission_classes = [AllowAny]
```
Comment (roman-tiukh) [inline stats/views.py]: again?)
it is impractical to open access to data to everyone in a commercial product without reason

File: stats/serializers.py [hunk 6]
```diff
@@ -1,17 +1,26 @@
...
 
+from business_register.models.company_models import CompanyType
 from business_register.models.kved_models import Kved
 from business_register.serializers.kved_serializers import KvedDetailSerializer
 
 
 class TopKvedSerializer(serializers.ModelSerializer):
     kved = serializers.SerializerMethodField()
-    count_kved = serializers.IntegerField()
+    count_kveds = serializers.IntegerField()
```
Comment (roman-tiukh) [inline stats/serializers.py:10]: count_companies more fit here

File: business_register/serializers/company_serializers.py (hunk 3 above)
Comment (IegorSoboliev) [reply business_register/serializers/company_serializers.py]: restored

File: stats/views.py (hunk 4 above)
Comment (IegorSoboliev) [reply stats/views.py]: done

File: stats/serializers.py (hunk 6 above)
Comment (IegorSoboliev) [reply stats/serializers.py]: renamed

File: stats/views.py (hunk 5 above)
Comment (IegorSoboliev) [reply stats/views.py]: fixed

**Pull request merged**